sdk/java/pulumi/src/main/java/com/pulumi/serialization/internal/Converter.java

```diff
@@ -421,24 +421,25 @@ private Object tryConvertObjectInner(
                             setters.get(name).getName(),
                             Nullable.class.getTypeName()
                     ));
-                }
-                try {
-                    var convertedArgument = argument == null ? null : tryConvertObjectInner(
-                            String.format("%s(%s)", targetType.getTypeName(), name),
-                            argument,
-                            TypeShape.extract(extractSetterParameter(setters.get(name)))
-                    );
-                    setters.get(name).invoke(builder, convertedArgument);
-                } catch (IllegalArgumentException e) {
-                    throw new IllegalStateException(String.format(
-                            "Error invoking setter '%s' (on '%s'), setter parameters: '%s', argument type: '%s'",
-                            name, targetType.getTypeName(),
-                            Arrays.toString(setters.get(name).getParameterTypes()),
-                            argument == null ? "null" : argument.getClass()
-                    ), e);
-                } catch (IllegalAccessException | InvocationTargetException e) {
-                    var exMsg = e.getMessage() == null ? e.getClass().getSimpleName() : e.getMessage();
-                    throw new IllegalStateException(String.format("Unexpected exception: %s", exMsg), e);
+                } else {
```

Comment:
@t0yv0  adding this else here, has a slightly different semantics then https://github.com/pulumi/pulumi-java/pull/840, but both should work in practice

the difference is, should we call the `@Nullable` setters, to make sure we don't have dirty mutable state issues, but on the other hand this else only helps with that, but do not guarantee lack of dirty state